lib/src/content/components/main_content/wolt_modal_sheet_main_content.dart

```diff
@@ -13,15 +13,104 @@ class WoltModalSheetMainContent extends StatelessWidget {
   final SliverWoltModalSheetPage page;
   final WoltModalType woltModalType;
   final WoltModalSheetScrollAnimationStyle scrollAnimationStyle;
+  final WoltModalSheetRoute route;
+  final VoidCallback? onModalDismissedWithDrag;
+  final GlobalKey modalContentKey;
 
   const WoltModalSheetMainContent({
     required this.scrollController,
     required this.pageTitleKey,
     required this.page,
     required this.woltModalType,
     required this.scrollAnimationStyle,
+    required this.route,
+    this.onModalDismissedWithDrag,
+    required this.modalContentKey,
     Key? key,
   }) : super(key: key);
+  RenderBox get _renderBox =>
+      modalContentKey.currentContext!.findRenderObject()! as RenderBox;
+
+  double get _childHeight => _renderBox.size.height;
+  bool get _shouldDragContent =>
+      (page.contentDraggable ?? false) && woltModalType is WoltBottomSheetType;
+
+  AnimationController get _animationController => route.animationController!;
+
+  double get _minFlingVelocity => woltModalType.minFlingVelocity;
+
+  bool get _isDismissUnderway =>
+      _animationController.status == AnimationStatus.reverse;
+
+  bool get _isDismissed => _animationController.isDismissed;
+
+  WoltModalDismissDirection? get _dismissDirection =>
+      woltModalType.dismissDirection;
+
+  void _handleVerticalDragUpdate(
+      DragUpdateDetails details, BuildContext context) {
+    if (_isDismissUnderway || _isDismissed) {
+      return;
+    }
+
+    final deltaDiff = details.primaryDelta! / _childHeight;
+    _animationController.value -= deltaDiff;
+
+    if (_dismissDirection == WoltModalDismissDirection.down) {
+      _animationController.value -= deltaDiff;
+    } else if (_dismissDirection == WoltModalDismissDirection.up) {
+      _animationController.value += deltaDiff;
+    }
+  }
+
+  void _handleVerticalDragEnd(BuildContext context, DragEndDetails details) {
```

Comment:
Why don't you wrap the main content with `WoltModalSheetContentGestureDetector` instead of repeating the code here?